### PR TITLE
Create eslint-config-target-es JS package

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

This package is an ESLint sharable config that will enable rules from
eslint-plugin-es based on the browserslist query.

The intention is to use this to replace the old "validate-es5" checks we
had before we dropped IE11 support.

Note this PR doesn't start using it in any builds yet. That's left for a followup.
Configuring a mirror repo and publishing is also left for a followup.

#### Jetpack product discussion

See #20817 and p1HpG7-c8n-p2.

This could help avoid regressions in browser support in the course of p9dueE-3MG-p2.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Try it out. For example, you might go to `projects/plugins/jetpack` and `pnpx validate-es _inc/build _inc/blocks`.

### CODE REVIEW

**Review by kraftbj — APPROVED** [review]
This is handy; thanks for putting it together.

**Comment by anomiex** [comment]
Needs re-approval after a merge conflict.

**Review by sdixon194 — APPROVED** [review]

**Comment by anomiex** [comment]
r234965-wpcom

**Pull request merged**
